hopsworks-api/src/main/java/io/hops/hopsworks/api/project/MetadataService.java

```diff
@@ -585,50 +586,46 @@ private Response mutateMetadata(Users user, String metaObj, MetadataOp op)
             build();
 
   }
-
-  @POST
-  @Path("attachSchemaless")
-  @Consumes(MediaType.APPLICATION_JSON)
-  @AllowedProjectRoles({AllowedProjectRoles.DATA_SCIENTIST, AllowedProjectRoles.DATA_OWNER})
-  public Response attachSchemalessMetadata(String metaObj) throws DatasetException, MetadataException {
-    processSchemalessMetadata(metaObj, false);
-    return noCacheResponse.getNoCacheResponseBuilder(Response.Status.ACCEPTED).build();
-  }
-
-  @POST
-  @Path("detachSchemaless")
+  
+  @PUT
+  @Path("schemaless/{xattrName}/{path: .+}")
   @Consumes(MediaType.APPLICATION_JSON)
   @AllowedProjectRoles({AllowedProjectRoles.DATA_SCIENTIST, AllowedProjectRoles.DATA_OWNER})
-  public Response detachSchemalessMetadata(String metaObj) throws DatasetException, MetadataException {
-    processSchemalessMetadata(metaObj, true);
-    return noCacheResponse.getNoCacheResponseBuilder(Response.Status.ACCEPTED).build();
-  }
-
-  private void processSchemalessMetadata(String metaObj, boolean detach)
-    throws DatasetException, MetadataException {
-    JsonObject jsonObj = Json.createReader(new StringReader(metaObj)).
-      readObject();
-    if (!jsonObj.containsKey("path")) {
-      throw new IllegalArgumentException("missing path field in metaObj");
-    }
+  public Response attachSchemalessMetadata(@Context SecurityContext sc,
+      @PathParam("xattrName") String xattrName, @PathParam("path") String path,
+      String metaObj) throws DatasetException,
+      MetadataException {
+    Users user = jWTHelper.getUserPrincipal(sc);
   
-    String inodePath = jsonObj.getString("path");
-    if (detach) {
-      metadataController.removeSchemaLessMetadata(inodePath);
-      return;
+    if (metaObj.length() > 13500 || xattrName.length() > 255) {
+      throw new MetadataException(RESTCodes.MetadataErrorCode.METADATA_MAX_SIZE_EXCEEDED, Level.FINE);
     }
+    metadataController.addSchemaLessMetadata(user, path, xattrName,
+        metaObj);
+    return noCacheResponse.getNoCacheResponseBuilder(Response.Status.CREATED).build();
```

Comment:
@maismail Remove the `nocacheResponse`. Use Response as we do in `JobsResource`

Reply:
done